tests/e2e/tests/test_storage_efs.py

```diff
@@ -63,6 +105,134 @@ def test_pvc_with_volume(
+            namespace=DEFAULT_USER_NAMESPACE,
+        )
+        arguments = {"mount_path": MOUNT_PATH, "claim_name": CLAIM_NAME}
+
+        # Write Pipeline Run
+        write_run_id = kfp_client.create_run_from_pipeline_func(
+            write_to_volume_pipeline,
+            experiment_name=experiment_name,
+            namespace=DEFAULT_USER_NAMESPACE,
+            arguments=arguments,
+        ).run_id
+        print(f"write_pipeline run id is {write_run_id}")
+        wait_for_kfp_run_succeeded_from_run_id(kfp_client, write_run_id)
+
+        # Read Pipeline Run
+        read_run_id = kfp_client.create_run_from_pipeline_func(
+            read_from_volume_pipeline,
+            experiment_name=experiment_name,
+            namespace=DEFAULT_USER_NAMESPACE,
+            arguments=arguments,
+        ).run_id
+        print(f"read_pipeline run id is {read_run_id}")
+        wait_for_kfp_run_succeeded_from_run_id(kfp_client, read_run_id)
+
+
+class TestEFS_Dynamic:
+    @pytest.fixture(scope="class")
+    def setup_dynamic(self, metadata, kustomize, port_forward, dynamic_provisioning):
+        # Disable caching in KFP
+        # By default KFP will cache previous pipeline runs and subsequent runs will skip cached steps
+        # This prevents artifacts from being uploaded to s3 for subsequent runs
+        patch_body = unmarshal_yaml(DISABLE_PIPELINE_CACHING_PATCH_FILE)
+        k8s_admission_registration_api_client = (
+            create_k8s_admission_registration_api_client(cluster, region)
+        )
+        k8s_admission_registration_api_client.patch_mutating_webhook_configuration(
+            "cache-webhook-kubeflow", patch_body
+        )
+
+        metadata_file = metadata.to_file()
+        print(metadata.params)  # These needed to be logged
+        print("Created metadata file for TestEFS_Dynamic", metadata_file)
+
+    def test_pvc_with_volume_dynamic(
+        self,
+        metadata,
+        setup_dynamic,
+        kfp_client,
+        account_id,
+        dynamic_provisioning,
+    ):
+        driver_list = subprocess.check_output("kubectl get csidriver".split()).decode()
+        assert "efs.csi.aws.com" in driver_list
+
+        pod_list = subprocess.check_output("kubectl get pods -A".split()).decode()
+        assert "efs-csi-controller" in pod_list
+
+        sa_account = subprocess.check_output(
+            "kubectl describe -n kube-system serviceaccount efs-csi-controller-sa".split()
+        ).decode()
+        assert f"arn:aws:iam::{account_id}:role" in sa_account
+
+        fs_id = dynamic_provisioning["file_system_id"]
+        assert "fs-" in fs_id
+
+        CLAIM_NAME = dynamic_provisioning["efs_claim_dyn"]
+        claim_list = subprocess.check_output("kubectl get pvc -A".split()).decode()
+        assert CLAIM_NAME in claim_list
+
+        claim_entry = subprocess.check_output(
+            f"kubectl get pvc -n {DEFAULT_USER_NAMESPACE} {CLAIM_NAME} -o=jsonpath='{{.status.phase}}'".split()
```

Comment:
fyi: we are trying to use the k8s python library for most of these kubectl things https://github.com/awslabs/kubeflow-manifests/blob/main/tests/e2e/utils/custom_resources.py

Reply:
noted, let me fix that.